json/decode.go

```diff
@@ -300,9 +300,9 @@ func (d decoder) decodeFromString(b []byte, p unsafe.Pointer, decode decodeFunc)
 		return decode(d, b, p)
 	}
 
-	v, b, _, err := parseStringUnquote(b, nil)
+	v, b, _, err := parseStringUnquote(b, nil, d.flags)
```

Comment:
Would it be interesting to make these `parse*` functions methods of `decoder` as well?

Reply:
See f0fbb8de7554450c31082cdd0c7875a8c32c039b